cocos/audio/android/UrlAudioPlayer.cpp

```diff
@@ -0,0 +1,339 @@
+/****************************************************************************
+Copyright (c) 2016 Chukong Technologies Inc.
+
+http://www.cocos2d-x.org
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+****************************************************************************/
+
+#define LOG_TAG "UrlAudioPlayer"
+
+#include "audio/android/UrlAudioPlayer.h"
+#include "audio/android/ICallerThreadUtils.h"
+#include "audio/android/AssetFd.h"
+
+#include <math.h>
+#include <algorithm> // for std::find
+
+namespace cocos2d { namespace experimental {
+
+static std::vector<UrlAudioPlayer *> __allPlayers;
+
+static std::once_flag __onceFlag;
+static int __instanceCount = 0;
```

Comment:
i think there are better options to do:

1. make them as static member variables of UrlAudioPlayer
2. define name int to an anonymous namespace without static